src/app/features/course-info/course-info.component.ts

```diff
@@ -1,19 +1,47 @@
-import { Component, EventEmitter, Input, Output } from "@angular/core";
-import { ButtonTypes } from "@app/shared/types/button.type";
-import { CourseInfo } from "@app/shared/types/course.model";
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { CoursesStoreService } from '@app/services/courses-store.service';
+import { MappingService } from '@app/services/mapping.service';
+import { ButtonTypes } from '@app/shared/types/button.type';
+import { CourseInfo } from '@app/shared/types/course.model';
+import { combineLatest, Subscription } from 'rxjs';
 
 @Component({
-  selector: "app-course-info",
-  templateUrl: "./course-info.component.html",
-  styleUrls: ["./course-info.component.scss"],
+  selector: 'app-course-info',
+  templateUrl: './course-info.component.html',
+  styleUrls: ['./course-info.component.scss'],
 })
-export class CourseInfoComponent {
+export class CourseInfoComponent implements OnInit, OnDestroy {
   ButtonTypes = ButtonTypes;
+  course: CourseInfo | undefined = { id: '', title: '', duration: 0, description: '', authors: [], date: '' };
```

Comment:
Instead of initializing all fields with some dummy data, you can do this. Neat and tidy.
![image](https://github.com/user-attachments/assets/ad479e18-741f-43d8-9ab4-6e43e37dd136)

Reply:
Hmm much nicer, thank you!